sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -0,0 +1,273 @@
+    this.response.statusCode = 200;
+    switch (dataType) {
+      case "json":
+        this.response.setHeader("Content-Type", "application/json; charset=utf-8");
+        break;
+      case "text":
+        this.response.setHeader("Content-Type", "text/plain; charset=utf-8");
+        break;
+      default:
+        this.response.setHeader("Content-Type", "application/octet-stream");
+        break;
+    }
+    this.response.end(data ?? "");
+  }
+  public fail(code: 400 | 401 | 500, detail?: string): void {
+    this.response.statusCode = code;
+    this.response.end(detail ?? "");
+  }
+}
+
+export class CloudEventsDispatcher {
+  private readonly _dumpRequest: boolean;
+  private readonly _allowedOrigins: string[];
+  constructor(
+    private hub: string,
+    allowedEndpoints: string[],
+    private eventHandler?: WebPubSubEventHandlerOptions
+  ) {
+    this._dumpRequest = eventHandler?.dumpRequest ?? false;
+    this._allowedOrigins = allowedEndpoints.map((endpoint) =>
+      endpoint === "*" ? "*" : new URL(endpoint).host
+    );
+  }
+
+  public processValidateRequest(req: IncomingMessage, res: ServerResponse): boolean {
+    if (req.headers["webhook-request-origin"]) {
+      res.setHeader("WebHook-Allowed-Origin", this._allowedOrigins);
+      res.end();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public async processRequest(
+    request: IncomingMessage,
+    response: ServerResponse
+  ): Promise<boolean> {
+    // check if it is a valid WebPubSub cloud events
+    var eventType = this.tryGetWebPubSubEvent(request);
+    if (eventType === undefined) {
+      return false;
+    }
+
+    // check if hub matches
+    if (request.headers["ce-hub"] !== this.hub) {
+      return false;
+    }
+
+    // No need to read body if handler is not specified
+    switch (eventType) {
+      case EventType.Connect:
+        if (!this.eventHandler?.handleConnect) {
+          response.statusCode = 401;
+          response.end("Connect event handler is not configured.");
+          return true;
+        }
+        break;
+      case EventType.Connected:
+        if (!this.eventHandler?.onConnected) {
+          response.end();
+          return true;
+        }
+        break;
+      case EventType.Disconnected:
+        if (!this.eventHandler?.onDisconnected) {
+          response.end();
+          return true;
+        }
+        break;
+      case EventType.UserEvent:
+        if (!this.eventHandler?.handleUserEvent) {
+          response.end();
+          return true;
+        }
+        break;
+      default:
+        throw new Error(`Unknown EventType ${eventType}`);
+    }
+
+    var eventRequest = await this.convertHttpToEvent(request);
+    const receivedEvent = HTTP.toEvent(eventRequest);
+
+    if (this._dumpRequest) {
+      console.log(receivedEvent);
+    }
+
+    switch (eventType) {
+      case EventType.Connect: {
+        var handler = new DefaultConnectResponseHandler(response);
```

Comment:
There are multiple uses of `var` throughout. Can we replace them with either `const` or `let` as appropriate?

Reply:
sure